docs/source/upcoming_release_notes/348-ECS-6433_heuristic_loading.rst

```diff
@@ -0,0 +1,24 @@
+348 ECS-6433 heuristic loading
+#################
+
+API Breaks
+----------
+- N/A
+
+Features
+--------
+- Updated happi's load() function to support searching and loading in one user-friendly expression. The load function gathers the results of each search term individually and loads the union of the results.
+
+Bugfixes
+--------
+- N/A
+
+Maintenance
+-----------
+- Cleaned up code by removing stray print() statements.
+- Added unit tests for the new feature.
+- In the load function of cli.py removed `term=[]` and replaced with `[item]` for readability.
```

Comment:
These are maybe too granular for release notes (the better side of the problem to have!)  Also the last change noted here was made then reverted over the course of the PR.

I think there actually aren't any Maintenance - level contributions to call out in this PR:
* There weren't print statements in happi before this PR 
* tests (should) come implicitly with any feature addition
* the `term` variable changes were added and removed within the course of this PR, and it's also a little too inside-baseball for release notes.